frontend/src/metabase/common/components/EntityPicker/types.ts

```diff
@@ -24,11 +25,31 @@ export type PickerStateItem<Item, Query> = {
 
 export type EntityPickerOptions = EntityPickerModalOptions;
 
-export type EntityTab<Model extends string> = {
+export type EntityPickerTabRenderProps<
+  Id extends SearchResultId,
+  Model extends string,
+  Item extends TypeWithModel<Id, Model>,
+> = {
+  onItemSelect: (item: Item) => void;
+};
+
+export type EntityPickerTabId = string;
+
+export type EntityPickerTab<
+  Id extends SearchResultId,
+  Model extends string,
+  Item extends TypeWithModel<Id, Model>,
+> = {
+  id: EntityPickerTabId;
   displayName: string;
-  element: JSX.Element;
+  render: (props: EntityPickerTabRenderProps<Id, Model, Item>) => JSX.Element;
   icon: IconName;
-  model: Model;
+  /**
+   * Recents & Search tabs don't have models associated with them - hence null
+   * (they provide the same models as the other tabs combined).
+   */
+  model: Model | null;
+  folderModels: Model[];
```

Comment:
This is to make `EntityPickerModal` aware of which items are folders (because it's only interested in folder clicks)